src/Tizen.Applications.Common/Tizen.Applications/EventSystem.cs

```diff
@@ -0,0 +1,172 @@
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// ev.EventReceived += EventReceivedCallback;
+        /// static void EventReceivedCallback(EventSystemArgs e)
+        /// {
+        ///     Console.WriteLine("Event Received ");
+        /// }
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<EventSystemArgs> EventReceived;
+
+        /// <summary>
+        /// The name of the event.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string EventName
+        {
+            get
+            {
+                return _eventName;
+            }
+        }
+        
+        /// <summary>
+        /// If true, the event is listening, otherwise false.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool Listening
+        {
+            get
+            {
+                return _listening;
+            }
+        }
+
+        /// <summary>
+        /// Listening event.
+        /// </summary>
+        /// <exception cref="System.InvalidOperationException">Thrown when portName is already used, when there is an I/O error.</exception>
+        /// <exception cref="System.ArgumentException">Thrown when there is an invalid parameter.</exception>
+        /// <exception cref="System.OutOfMemoryException">Thrown when out of memory.</exception>
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// ev.EventReceived += EventReceivedCallback;
+        /// ev.Listen();
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public void Listen()
+        {
+            _eventCallBack = (string eventName, IntPtr eventData, IntPtr userData) =>
+            {   
+                EventSystemArgs args = new EventSystemArgs()
+                {
+                    EventSystemData = new Bundle(new SafeBundleHandle(eventData, false))
+                };                
+                EventReceived?.Invoke(this, args);
+            };
+
+            Interop.EventSystem.AddEventHandler(_eventName, _eventCallBack, IntPtr.Zero, out _handler);
+            _listening = true;
+        }
+
+        /// <summary>
+        /// Stop listening event.
+        /// </summary>
+        /// <exception cref="System.InvalidOperationException">Thrown when event listening is already stopped or fail to remove handler</exception>
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// ev.EventReceived += EventReceivedCallback;
+        /// ev.Listen();
+        /// ev.StopListening();
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public void StopListening()
```

Comment:
I think this API may be removed also. We can remove the native handler when all the event handlers are removed.